Models/Movie.swift

```diff
@@ -7,31 +7,39 @@
 
 import Foundation
 
-struct Movies: Codable {
+struct Movies: Decodable {
     var results: [Movie]
 }
 
-struct Movie: Codable {
+struct Movie: Decodable {
+    enum Popularity {
+        case high, low
+    }
+    
     let id: Int
-    let posterPath: String
+    let posterPath: URL?
     let overview: String
     let title: String
     let voteAverage: Int
+    let popularity: Popularity
     
     enum CodingKeys: String, CodingKey {
-        case id
+        case id, overview, title, reviewsScore
         case posterPath = "poster_path"
-        case overview
-        case title
         case voteAverage = "vote_average"
     }
     
     init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: CodingKeys.self)
         id = try values.decode(Int.self, forKey: .id)
-        posterPath = try values.decode(String.self, forKey: .posterPath)
+        
+        let posterPathString = try values.decode(String.self, forKey: .posterPath)
```

Comment:
From the API docs [here](https://developers.themoviedb.org/3/movies/get-now-playing), `poster_path` can be String or Null so we should use the optional decoding here
```suggestion
        let posterPathString = try values.decodeIfPresent(String.self, forKey: .posterPath)
```

Reply:
So, it's going to be

if let posterPathString = try values.decodeIfPresent(String.self, forKey: .posterPath) {
            posterPath = URL(string: "https://image.tmdb.org/t/p/w500" + posterPathString)
        } else {
            posterPath = nil
        }